src/machine/machinectl.c

```diff
@@ -117,12 +118,63 @@ static int compare_machine_info(const void *a, const void *b) {
         return strcmp(x->name, y->name);
 }
 
+static void clean_machine_info(MachineInfo *machines, size_t n_machines) {
+        size_t i;
+
+        if (!machines || !n_machines)
+                return;
+
+        for (i = 0; i < n_machines; i++) {
+                free(machines[i].os);
+                machines[i].os = NULL;
```

Comment:
no need to initialize the .os field to NULL if we free it anyway right after. 

(Also, note that we have mfree() which frees memory and returns NULL, so that you could shorten the above to: `machines[i].os = mfree(machines[i].os)`...)